spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -400,6 +401,23 @@ class DeltaAnalysis(session: SparkSession)
           throw DeltaErrors.notADeltaTableException("RESTORE")
       }
 
+    // Resolve as a resolved table if the path is for delta table. For non delta table, we keep the
+    // path and pass it along.
+    case u: UnresolvedPath =>
+      val table = getPathBasedDeltaTable(u.path)
+      val tableExists = try {
+        table.tableExists
+      } catch {
+        case NonFatal(e) => false
+      }
+      if (!tableExists) {
+        u
+      } else {
+        val catalog = session.sessionState.catalogManager.currentCatalog.asTableCatalog
```

Comment:
See comment https://github.com/delta-io/delta/pull/2013#discussion_r1317756751

Reply:
Updated to use the catalog extracted from table.
Thanks for spotting it!